ucp/_libs/core.pyx

```diff
@@ -75,52 +76,62 @@ async def exchange_peer_info(endpoint, msg_tag, ctrl_tag, guarantee_msg_order):
     }
 
 
-# "1" is shutdown, currently the only opcode.
-cdef struct CtrlMsgData:
-    int64_t op  # The control opcode, currently the only opcode is "1" (shutdown).
-    int64_t close_after_n_recv  # Number of recv before closing
+class CtrlMsg:
+    """Implementation of control messages
 
+    For now we have one opcode `1` which means shutdown.
+    The opcode takes `close_after_n_recv`, which is the number of
+    messages to receive before the worker should close.
+    """
 
-cdef class CtrlMsg:
-    cdef CtrlMsgData data
+    fmt = "QQ"
+    nbytes = struct.calcsize(fmt)
 
+    @staticmethod
+    def serialize(opcode, close_after_n_recv):
+        return struct.pack(CtrlMsg.fmt, int(opcode), int(close_after_n_recv))
```

Comment:
I know this is already merged but wanted to say thanks to @madsbk for this work.  I raised some concerns in a [previous PR](https://github.com/rapidsai/ucx-py/pull/418#discussion_r389231002) (probably wrong) and I appreciate you taking a look.

Reply:
Yeah Mads is right that it should have been `"i"` in that context. Though it seems we have dropped that value since.